src/pages/question/PersonaQuestion.tsx

```diff
@@ -1,45 +1,70 @@
-import React, { ReactElement, useState } from 'react';
-import { useDispatch, useSelector } from 'react-redux';
+import Icon from 'components/icon';
+import React, { ReactElement, SyntheticEvent, useState } from 'react';
 import { useHistory } from 'react-router-dom';
 
 import { Choice } from '../../entity';
-import { personaQuestionSelector,questionsAction } from '../../store/personaQuestionsSlice';
 import Card from './card';
-import * as S from './styled';;
+import Loading from './loading';
+import questions from './questions';
+import * as S from './styled';
 
 export default function PersonaQuestionPage(): ReactElement {
-  const dispatch = useDispatch();
-  const { questions } = useSelector(personaQuestionSelector);
   const history = useHistory();
   const [currentIdx, setCurrentIdx] = useState(0);
-  const quesitonLen = questions.length;
+  const [answer, setAnswer] = useState<Choice[]>([]);
+  const [isLoading, setLoading] = useState(false);
   const { title, choices } = questions[currentIdx];
-  const progressVal = (100 / quesitonLen) * (currentIdx + 1);
+  const questionLen = questions.length;
+  const questionNum = currentIdx + 1 < 10 ? '0' + (currentIdx + 1).toString() : currentIdx + 1;
+  const progressVal = (100 / questionLen) * (currentIdx + 1);
 
-  const handleCardClick = () => {
-    if (currentIdx < quesitonLen - 1) {
+  const getChoice = (index: number) => {
+    return choices[index];
+  };
+
+  const handleCardClick = (e: SyntheticEvent<HTMLButtonElement>) => {
+    const choiceIdx: string | undefined =
+      e.currentTarget.dataset.index != undefined ? e.currentTarget.dataset.index : '';
+    setAnswer(() => [...answer, getChoice(parseInt(choiceIdx))]);
+    if (currentIdx < questionLen - 1) {
       setCurrentIdx(currentIdx + 1);
-    } else {
-      history.push('/persona/result');
+    }
+    if (currentIdx >= questionLen - 1) {
+      setLoading(true);
+      // history.push('/persona/result');
     }
   };
 
-  const renderCardList = choices.map((item: Choice) => {
-    return (
-      <Card key={item.uid} uid={item.uid} contents={item.contents} onClick={handleCardClick}></Card>
-    );
+  const handleBackButtonClick = () => {
+    setAnswer(() => answer.splice(answer.length - 1, 1));
+    setCurrentIdx(currentIdx - 1);
+  };
+
+  const renderCardList = choices.map((item: Choice, index: number) => {
+    return <Card key={index} uid={index} contents={item.contents} onClick={handleCardClick}></Card>;
   });
 
-  return (
-    <S.Container>
-      <S.BackButton>-</S.BackButton>
-      <S.QuestionID>{`${currentIdx + 1}/${quesitonLen}`}</S.QuestionID>
-      <S.Title>{title}</S.Title>
-      <S.CardDiv>{renderCardList}</S.CardDiv>
-      <S.Container>PersonaQuestionPage</S.Container>;
-      <S.ProgressContainer>
-        <S.ProgressComplete barWidth={progressVal}></S.ProgressComplete>
-      </S.ProgressContainer>
-    </S.Container>
-  );
+  if (isLoading) {
+    return <Loading />;
+  } else {
+    return (
```

Comment:
P4;

early-return 형태가 가독성이 더 좋지 않을까 생각합니다.
이미 if문에 return이 있기 때문에, else는 불필요합니다.
```suggestion
  if (isLoading) {
    return <Loading />;
  }
  return (
```

Reply:
그렇네요 제안 감사합니다!